sched/semaphore/sem_post.c

```diff
@@ -85,19 +87,54 @@ int nxsem_post_slow(FAR sem_t *sem)
 
   flags = enter_critical_section();
 
-  /* Check the maximum allowable value */
+  if (mutex)
+    {
+      /* Mutex post from interrupt context is not allowed */
+
+      DEBUGASSERT(!up_interrupt_context());
+
+      /* Lock the mutex for us by setting the blocking bit */
+
+      mholder = atomic_fetch_or(NXSEM_MHOLDER(sem), NXSEM_MBLOCKS_BIT);
+
+      /* Mutex post from another thread is not allowed, unless
```

Comment:
Maybe I don't understand the use case well enough, but one option IF a non-owner thread tries to post from nxmutex_reset, it could just take ownership of the mutex there and then post. But what is the use case for this I wonder ? Sounds like asking for trouble tbh.

Reply:
Going around the reset again.... I DO want to remove it. But I don't understand why it is there in that serial driver, and I implemented it in the way that it works as before. So I don't want to remove it before someone removes that from the serial driver.

Reply:
Yup, it's the part that has me most baffled. Just because I don't understand what it's supposed to do. I do get what it did before. Keeping the existing functionality as close as possible is a fair choice.

Reply:
Yes, sticking to that for now... all my attempts to remove it caused havoc, especially with USB CDCACM (in the drone mavlink over USB). Replicating the functionality of nxmutex_reset just does the trick for now. Eventually I got fed up with debugging USB and serial drivers. There has to be some design flaw in the serial if one needs to "reset" mutexes, I just couldn't wrap my head around that yet, I am currently working on a bit too many things at the same time....

Reply:
Let's not try to remove it here, instead make a coherent implementation for nxmutex_reset.